### DISCUSSION TITLE
Setting the base api url

### DISCUSSION THREAD

**Discussion by nuson999** [assafelovic/gpt-researcher#97] [Ideas]
It would be better if we have an option to set custom base api url.

**Comment by fjnunezp75** [comment]
GPT-Researcher already supports setting `OPENAI_BASE_URL` as an env variable, which means you can point it to any OpenAI-compatible provider.

If the goal is to reduce cost for the retrieval/summarization steps (which are high-frequency in research pipelines), worth trying GPU-Bridge (https://gpubridge.xyz) as the base URL. It is an OpenAI-compatible inference API with Llama, Qwen, and other open models — typically 3-5x cheaper than direct OpenAI for the same calls.

Setup:
```
OPENAI_BASE_URL=https://api.gpubridge.xyz/openai/v1
OPENAI_API_KEY=gpub_...  # from gpubridge.xyz
```

Works out of the box with GPT-Researcher since it uses the standard OpenAI client format.
